src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/IcraEligibilityViewWorkExperienceReference.vue

```diff
@@ -48,14 +129,109 @@ export default defineComponent({
   data() {
     return {
       icraEligibilityStatus: {} as Components.Schemas.ICRAEligibilityStatus,
+      resending: false as boolean,
     };
   },
   async mounted() {
-    this.icraEligibilityStatus = (await getIcraEligibilityStatus(this.icraEligibilityId))?.data || {};
+    this.icraEligibilityStatus =
+      (await getIcraEligibilityStatus(this.icraEligibilityId))?.data || {};
```

Comment:
there is an endpoint to just grab the reference. I think it will simplify your selectedReference computed property. 

endpointRouteBuilder.MapGet("/api/icra/workReference/{workExperienceReferenceId}", async Task<Results<Ok<EmploymentReference>